tests/GUI/sortList.spec.ts

```diff
@@ -0,0 +1,55 @@
+import test, { Page } from "@playwright/test";
+import { LeftMenu } from "../../page-objects/leftMenu";
+import tasks from "../../resources/tasks-import.json";
+import { ApiHooks } from "../../api-utils/apiHooks";
+import { ProjectMainView } from "../../page-objects/projectMainView";
+
+let leftMenu: LeftMenu;
+let projectMainView: ProjectMainView;
+
+let page: Page;
+
+test.describe(
+  "UI tests for checking if sorting list works correct",
+  {
+    tag: ["@list", "@sorting"],
+    annotation: {
+      type: "issue",
+      description: "https://github.com/marian51/poc-playwright-typescript-gui-api/issues/85",
+    },
+  },
+  async () => {
+    test.beforeAll(async ({ browser, request }) => {
+      page = await browser.newPage();
+      for (const task of tasks) {
+        await ApiHooks.createNewTaskWithData(request, task);
+        console.log("Created task: " + task.name)
+      }
+    });
+
+    [
+      { key: "Due date", direction: "Ascending" },
+      { key: "Due date", direction: "Descending" },
+      { key: "Priority", direction: "Ascending" },
+      { key: "Priority", direction: "Descending" },
+      { key: "Task Name", direction: "Ascending" },
+      { key: "Task Name", direction: "Descending" },
+    ].forEach(({ key, direction }) => {
+      test(`Test of sorting tasks list by '${key}' column in '${direction}' direction`, async () => {
```

Comment:
Rozumiem, że to działa jak scenario outline. Może zamknąć ten array opcji do zmiennej nazwanej np. `sortOptions` lub `sortOptionsExamples` ?

```
const sortOptions = [
      { key: "Due date", direction: "Ascending" },
      { key: "Due date", direction: "Descending" },
      { key: "Priority", direction: "Ascending" },
      { key: "Priority", direction: "Descending" },
      { key: "Task Name", direction: "Ascending" },
      { key: "Task Name", direction: "Descending" },
];

sortOptions.forEach(({ key, direction }) => {
```

Można zastanowić się nad wyniesieniem takiej zmiennej do beforeAll, ale to wszystko tylko luźne sugestie, bo nie znam sposobów na implementowanie scenario outline w playwright.

Reply:
IMO jest spoko, pomysłowy sposob. Bardziej bym powiedział, że podobne do któregoś ze sposobów dostarczania danych do testów z selenium (tam tez była mapa/podobna struktura)

Reply:
No i myślę, że to ciekawy przypadek, pasuje bardzo do sortowania. W takim PoC chyba zawsze warto mieć jak najwięcej różnych sposobów/przykładów.

Reply:
To jest niemalże całkowicie ściągnięte z [➡️dokumentacji playwrighta](https://playwright.dev/docs/test-parameterize#before-and-after-hooks). W sumie można by wynieść tę tablicę obiektów z `key` i `direction` jako stałą wyżej, a nawet gdzieś do osobnej klasy, ale na razie ją tu zostawię - przypomina to trochę właśnie takie *scenario outline* z Cucumbera, jak leży blisko testu

Reply:
Sprawdziłem, i przeniesienie testu do bloku describe, żeby hooki `beforeAll` i `afterAll` były w scopie nad tym blokiem, nic nie dało 😐 Nadal w przypadku faila w którymś z testów z "outline" odpala się hook `afterAll` i po nim `beforeAll`. Próbowałem też zmienić hooki na `beforeEach` i `afterEach`, ale też nic to nie zmieniło